waiter/src/waiter/scheduler/kubernetes.clj

```diff
@@ -673,8 +670,14 @@
      :watch-state @watch-state})
 
   (validate-service [this service-id]
-    (let [run-as-user (retrieve-run-as-user this service-id)]
-      (authz/check-user authorizer run-as-user service-id))))
+    (let [{:strs [namespace run-as-user]} (retrieve-service-description this service-id)]
+      (authz/check-user authorizer run-as-user service-id)
+      ;; currently, if manually specified, the namespace *must* match the run-as-user
+      ;; (but we expect the common case to be falling back to the default)
+      (when (and (some? namespace)
+                 (not= namespace run-as-user))
```

Comment:
Can we move this validation to the `service_description.clj` validation method?

Reply:
Moved. Also moved the corresponding unit tests.